### PR TITLE
[issue-502] Dictionary exclude column should be dimension not measure

### PR DESCRIPTION
**Pull request by ManoharVanam**
Dictionary exclude column will be dimension

### CODE REVIEW

File: integration/spark/src/test/scala/org/apache/spark/sql/TestCarbonSqlParser.scala [hunk 1]
```diff
@@ -76,14 +76,290 @@ class TestCarbonSqlParser extends QueryTest {
...
     //No dictionary column names will be available in noDictionary list
+    assert(noDictionary.size == 1)
     assert(noDictionary.lift(0).get.equalsIgnoreCase("col2"))
   }
 
+  test("Test-DimAndMsrColsWithNoDictionaryFields1") {
+    val tableProperties = Map("DICTIONARY_EXCLUDE" -> "col1")
+    var fields: Seq[Field] = Seq[Field]()
+
+    var col1 = Field("col1", Option("Int"), Option("col1"), None, null, Some("columnar"))
```
Comment (gvramana) [inline integration/spark/src/test/scala/org/apache/spark/sql/TestCarbonSqlParser.scala]: move common part to methods

File: integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala [hunk 2]
```diff
@@ -637,24 +638,26 @@ class CarbonSqlParser()
...
-    var splittedCols: Array[String] = Array[String]()
+    var dictIncludedCols: Array[String] = Array[String]()
+    var dictExcludedCols: Array[String] = Array[String]()
 
-    // check any column need to be excluded
+    // get all included cols
     if (None != tableProperties.get("DICTIONARY_INCLUDE")) {
-      val dicIncludeCols: String = tableProperties.get("DICTIONARY_INCLUDE").get
-      splittedCols = dicIncludeCols.split(',')
+      dictIncludedCols = tableProperties.get("DICTIONARY_INCLUDE").get.split(',')
```
Comment (eason-lyx) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala]: i think we need to add trim for include/exclude cols.

Comment (ManoharVanam) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala]: yeah , need to add the same in the old flow also

File: integration/spark/src/test/scala/org/apache/spark/sql/TestCarbonSqlParser.scala (hunk 1 above)
Comment (ManoharVanam) [inline integration/spark/src/test/scala/org/apache/spark/sql/TestCarbonSqlParser.scala]: ok

**Comment by ravipesala** [comment]
LGTM

**Pull request merged**